src/main/java/com/bean00/RequestParser.java

```diff
@@ -7,18 +7,40 @@
 
 public class RequestParser {
     private BufferedReader in;
+    private static final int EXPECTED_NUMBER_OF_REQUEST_WORDS = 3;
+    private static final String EXPECTED_HTTP_VERSION = "HTTP/1.1";
 
     public RequestParser(BufferedReader in) {
         this.in = in;
     }
 
-    public Request parseRequest() throws IOException {
-        String requestLine = in.readLine();
-        String[] requestWords = requestLine.split("\\s");
+    public Request parseRequest() {
+        String requestLine;
+        try {
+            requestLine = in.readLine();
+        } catch (IOException e) {
+            throw new BadRequestHttpException();
```

Comment:
Maybe don't throw the `BadRequestHttpException` here (let the `IOException` fall through)